lib/ansible/parsing/mod_args.py

```diff
@@ -284,6 +287,12 @@ def parse(self):
 
         # walk the input dictionary to see we recognize a module name
         for (item, value) in iteritems(self._task_ds):
+            if item.count('.') == 2:
+                # This is a module from a galaxy role
+                gr = get_role_by_module_fqn(item)
+                if gr.module_file not in module_loader._module_cache:
+                    module_loader.add_module_by_file(item, gr.module_file)
```

Comment:
This could go away since the 'item' would be enough for plugin loader to get the right module